sonar-plugin-api/src/main/java/org/sonar/api/utils/Semaphores.java

```diff
@@ -27,8 +27,10 @@
  * A semaphore shared among all the processes that can connect to the central database.
  *
  * @since 3.4
+ * @deprecated in 5.2 as batch analyzer does not connect to database anymore
  */
 @ServerSide
+@Deprecated
 public interface Semaphores {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)